### PR TITLE
[quorum]Enable Quorum Tessera to use HashiCorp vault for keypair retrival

### PR DESCRIPTION
**Pull request by manikanta-darsi**
Signed-off-by: manikanta-darsi <manikanta.darsi@accenture.com>

**Changelog**
- Add logic to enable quorum tessera to use hashicorp vault for keypair retrival.

 

**Reviewed by**
@suvajit-sarkar @sownak @jagpreetsinghsasan 

 

**Linked issue**
#1822

### CODE REVIEW

File: platforms/quorum/charts/node_tessera/templates/configmap.yaml [hunk 1]
```diff
@@ -78,13 +78,22 @@ data:
...
+                  "hashicorpVaultSecretEngineName": "{{ $.Values.vault.secretengine }}",
+                  "hashicorpVaultSecretName": "{{ $.Values.vault.tmsecretpath }}",
+                  "hashicorpVaultPrivateKeyId": "privateKey",
+                  "hashicorpVaultPublicKeyId": "publicKey"
+              }
+          ]
       },
       "alwaysSendTo": []
-    }
+    }
```
Comment (sownak) [inline platforms/quorum/charts/node_tessera/templates/configmap.yaml:99]: Wrong EOF

File: platforms/quorum/charts/node_tessera/templates/deployment.yaml [hunk 2]
```diff
@@ -115,7 +115,7 @@ spec:
             jq -r 'if .errors then . else . end')
           validateVaultResponse "secret (${vault_secret_key})" "${LOOKUP_SECRET_RESPONSE}" "LOOKUPSECRETRESPONSE"
           nodekey=$(echo ${LOOKUP_SECRET_RESPONSE} | jq -r '.data.data["nodekey"]')
-          keystore=$(echo ${LOOKUP_SECRET_RESPONSE} | jq -r '.data.data["keystore"]' | base64 -d )
+          keystore=$(echo ${LOOKUP_SECRET_RESPONSE} | jq -r '.data.data["keystore"]' | base64 -d )
```
Comment (sownak) [inline platforms/quorum/charts/node_tessera/templates/deployment.yaml:118]: remove trailing spaces

File: platforms/quorum/charts/node_tessera/templates/deployment.yaml [hunk 3]
```diff
@@ -195,43 +182,71 @@ spec:
...
+            echo "Getting secrets from Vault Server: ${VAULT_ADDR}"
+            # Login to Vault and so I can get an approle token
+            VAULT_CLIENT_TOKEN=$(curl -sS --request POST ${VAULT_ADDR}/v1/auth/${KUBERNETES_AUTH_PATH}/login \
+              -H "Content-Type: application/json" \
+              -d '{"role":"'"${VAULT_APP_ROLE}"'","jwt":"'"${KUBE_SA_TOKEN}"'"}' | \
+              jq -r 'if .errors then . else .auth.client_token end')
+            validateVaultResponse 'vault login token' "${VAULT_CLIENT_TOKEN}"
+            
+            # required for tessera vault integration
+            export HASHICORP_TOKEN=${VAULT_CLIENT_TOKEN};
```
Comment (sownak) [inline platforms/quorum/charts/node_tessera/templates/deployment.yaml:225]: Why not used this directly in line 218 above?

File: platforms/quorum/charts/node_tessera/templates/deployment.yaml [hunk 4]
```diff
@@ -257,4 +272,4 @@ spec:
         accessModes: [ "ReadWriteOnce" ]
         resources:
           requests:
-            storage: {{ .Values.storage.dbstorage }}
+            storage: {{ .Values.storage.dbstorage }}
```
Comment (sownak) [inline platforms/quorum/charts/node_tessera/templates/deployment.yaml:275]: Add EOF

File: platforms/quorum/configuration/roles/helm_component/templates/crypto_tessera.tpl [hunk 5]
```diff
@@ -18,12 +18,13 @@ spec:
...
       pullSecret: regcred
     vault:
       address: {{ vault.url }}
+      secretengine: {{ vault.secret_path | default('secretsv2') }}      
       authpath: quorum{{ org_name }}
       keyprefix: {{ component_ns }}/crypto
       role: vault-role
       serviceaccountname: vault-auth
-      tmprefix: {{ vault.secret_path | default('secretsv2') }}/data/{{ component_ns }}/crypto
+      tmprefix: {{ vault.secret_path | default('secretsv2') }}/data/{{ component_ns }}/crypto
```
Comment (sownak) [inline platforms/quorum/configuration/roles/helm_component/templates/crypto_tessera.tpl:30]: EOF

File: platforms/quorum/configuration/roles/helm_component/templates/tessera.tpl [hunk 6]
```diff
@@ -44,16 +45,16 @@ spec:
       lock: {{ peer.lock | lower }}
       ports:
         rpc: {{ peer.rpc.port }}
-{% if network.config.consensus == 'raft' %}
```
Comment (sownak) [inline platforms/quorum/configuration/roles/helm_component/templates/tessera.tpl:47]: Why is this IF removed if raft is not needed, remove it entirely

File: platforms/quorum/configuration/roles/helm_component/templates/tessera.tpl [hunk 7]
```diff
@@ -91,4 +92,4 @@ spec:
     storage:
       storageclassname: {{ storageclass_name }}
       storagesize: 1Gi
-      dbstorage: 1Gi
+      dbstorage: 1Gi
```
Comment (sownak) [inline platforms/quorum/configuration/roles/helm_component/templates/tessera.tpl:95]: EOF

**Review by sownak — CHANGES_REQUESTED** [review]

File: platforms/quorum/charts/node_tessera/templates/configmap.yaml (hunk 1 above)
Comment (manikanta-darsi) [reply platforms/quorum/charts/node_tessera/templates/configmap.yaml]: done

File: platforms/quorum/charts/node_tessera/templates/deployment.yaml (hunk 2 above)
Comment (manikanta-darsi) [reply platforms/quorum/charts/node_tessera/templates/deployment.yaml]: done

File: platforms/quorum/charts/node_tessera/templates/deployment.yaml (hunk 4 above)
Comment (manikanta-darsi) [reply platforms/quorum/charts/node_tessera/templates/deployment.yaml]: added.

File: platforms/quorum/configuration/roles/helm_component/templates/crypto_tessera.tpl (hunk 5 above)
Comment (manikanta-darsi) [reply platforms/quorum/configuration/roles/helm_component/templates/crypto_tessera.tpl]: added.

File: platforms/quorum/configuration/roles/helm_component/templates/tessera.tpl (hunk 6 above)
Comment (manikanta-darsi) [reply platforms/quorum/configuration/roles/helm_component/templates/tessera.tpl:47]: removed.

File: platforms/quorum/charts/node_tessera/templates/deployment.yaml (hunk 3 above)
Comment (manikanta-darsi) [reply platforms/quorum/charts/node_tessera/templates/deployment.yaml]: added.

**Review by sownak — APPROVED** [review]

**Pull request merged**